libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridDeRefinement.cpp

```diff
@@ -57,21 +60,29 @@ meshkernel::UndoActionPtr CurvilinearGridDeRefinement::Compute()
     UInt nIndexOriginalGrid = 0;
     while (nIndexOriginalGrid < m_grid.NumN())
     {
-        UInt localNDeRefinement = 1;
+        int localNDeRefinement = 1;
         if (nIndexOriginalGrid >= m_lowerLeft.m_n && nIndexOriginalGrid < m_upperRight.m_n)
         {
-            localNDeRefinement = numNToDeRefine;
+            localNDeRefinement = nDeRefineFactor;
+            if (nIndexOriginalGrid + localNDeRefinement > m_upperRight.m_n)
+            {
+                localNDeRefinement = static_cast<int>(m_upperRight.m_n) - nIndexOriginalGrid;
+            }
         }
         deRefinedGrid.emplace_back(std::vector<Point>());
         deRefinedGrid.back().reserve(m_grid.NumM());
 
         UInt mIndexOriginalGrid = 0;
         while (mIndexOriginalGrid < m_grid.NumM())
         {
-            UInt localMDeRefinement = 1;
+            int localMDeRefinement = 1;
             if (mIndexOriginalGrid >= m_lowerLeft.m_m && mIndexOriginalGrid < m_upperRight.m_m)
             {
-                localMDeRefinement = numMToDeRefine;
+                localMDeRefinement = mDeRefineFactor;
+                if (mIndexOriginalGrid + localMDeRefinement > m_upperRight.m_m)
+                {
+                    localMDeRefinement = static_cast<int>(m_upperRight.m_m) - mIndexOriginalGrid;
+                }
```

Comment:
mIndexOriginalGrid is still type UInt, does this cause a compiler warning (especially in VS)